Automattic/crowdsignal-ui
# Fix edit title behavior

This PR points to a better UX regarding the project title.

Changing the title was triggering a full save which led to some jumpy experience on the editor.
The changeset includes a new state branch/action/selector/reducer:

- Reducer: `state.editor.title` (should it be more specific?)
- Action Type: `EDITOR_TITLE_SET`
- Action: `setEditorTitle( title ) => ( { type: EDITOR_TITLE_SET, title } )`
- Selector: `getEditorTitle()`

The title input change handler simply sets the `title` branch value.

`saveEditorContent` accepts the prop `title` among the options (3rd argument)

`useAutosave` have now a last minute call to `getEditorTitle()` to append to the `saveEditorContent` options argument

Changing the title DOES NOT enqueue an autosave, I could say either I didn't find a good way or a good reason for this so, it's by design.

It also addresses an issue where a newly created project could not be saved except by an autosave (`trying to read id from undefined (project)`)

## Test instructions
Checkout and head to the editor: https://crowdsignal.localhost:8000/project

- Edit the title. Once you *focus out* of the title input you should see "Save draft" became available
- Either add some content to the project (to actually enqueue an autosave) or hit "Save draft": project should save successfully and the title should be set
